Dockerfile

```diff
@@ -1,12 +1,11 @@
 FROM php:7.0-apache
 
 RUN apt-get update \
- && apt-get install -y git zlib1g-dev \
- && docker-php-ext-install zip \
- && a2enmod rewrite \
- && sed -i 's!/var/www/html!/var/www/public!g' /etc/apache2/apache2.conf \
- && mv /var/www/html /var/www/public \
- && curl -sS https://getcomposer.org/installer \
-  | php -- --install-dir=/usr/local/bin --filename=composer
+  && apt-get install -y git zlib1g-dev \
+  && docker-php-ext-install zip \
+  && a2enmod rewrite \
+  && sed -i 's!/var/www/html!/var/www/public!g' /etc/apache2/sites-enabled/000-default.conf \
```

Comment:
`docker-compose` mounts your local directory to `/var/www`, removing whatever is there in the base `Dockerfile`; we simply need to do the `sed` command to update the `VirtualHost` and everything is fine, as the `public` directory exists within the mount.